Add Plugin and Theme check sections

mattyrob opened the pull request:
This PR adds compatibility sections that check for known incompatible plugins active on an install and also for incompatible theme in use (Twenty Twenty-One)

This change is contingent on an update to the migrate API, for which there is already a PR.

https://github.com/ClassicPress/ClassicPress-APIs/pull/24

nylen commented on lib/admin-page.php:
```diff
@@ -449,6 +449,57 @@ function classicpress_check_can_migrate() {
+		echo "<tr>\n<td>$icon_preflight_pass</td>\n<td>\n";
+		printf( __(
+			/* translators: active theme name */
+			'It looks like you are using the <strong>%1$s</strong> theme. We believe it is compatible with ClassicPress.',
```
```suggestion
			'It looks like you are using the <strong>%1$s</strong> theme. We are not aware of any incompatibilities between this theme and ClassicPress.',
```

mattyrob replied:
Fixed

nylen commented on lib/admin-page.php:
```diff
@@ -449,6 +449,57 @@ function classicpress_check_can_migrate() {
 	echo "</td></tr>\n";
 
+	// Check: Conflicting Theme
+	$theme = wp_get_theme();
```
The parent and the child theme (if any) should both be checked.

mattyrob replied:
Updated patch should correct this now.

nylen commented on lib/admin-page.php:
```diff
@@ -449,6 +449,57 @@ function classicpress_check_can_migrate() {
+		echo "<tr>\n<td>$icon_preflight_fail</td>\n<td>\n";
+		/* translators: URI to list of conflucting plugins */
+		_e(
+			'We have detected one or more incompatible plugins that prevent migrating your site to ClassicPress. ',
```
Trailing space at the end of this string:

```suggestion
			'We have detected one or more incompatible plugins that prevent migrating your site to ClassicPress.',
```

Also, it is important to list the incompatible plugins.

nylen commented on lib/admin-page.php:
```diff
@@ -449,6 +449,57 @@ function classicpress_check_can_migrate() {
+		printf( __(
+			'Please visit our <a href="%s">Forum</a> for a list of known conflictong plugins.',
+			'switch-to-classicpress'
+		), 'https://docs.classicpress.net/installing-classicpress/#plugin-conflicts' );
```
Several issues here:

- This link does not point to the forums as the text indicates.
- Typo: `conflictong`
- The `#plugin-conflicts` part of this link does not work anymore.
- If the list of conflicting plugins comes from a documentation page instead of from the API, then the list will need to be maintained in two different places and it is only a matter of time until they become out of sync.

mattyrob replied:
Perhaps on screen reporting of conflicting plugins, see updated PR.

nylen commented:
I left some specific comments above. More generally, I'm a bit concerned that these checks will offer a false sense of security: since we can't catch all incompatibilities, we probably shouldn't show a green checkmark when the plugin doesn't find anything that's not a known incompatibility. I'm not sure the best way to handle this in the UI though - maybe only show this check if it fails?

mattyrob commented:
I have updated the PR as follows:

- 'Pass' icons changes to 'Warn' icons as passing information rather than assuring a smooth migration.
- Fixed trailing white space.
- Added checks for parent / child theme pairings.
- Display names of conflicting plugins on-screen rather than linking to docs / forum for improved UX and easier maintenance.

striebwj commented:
@mattyrob I think this is good to go now that we got the api update in?

mattyrob commented:
@striebwj The API being there means the plugin and theme checks are now operational, provided everyone is happy with the way the information is presented to users when migrating it can be pushed and released. I think all of the comments by @nylen have been addressed.
So, would just need another approval and we can push on.

striebwj commented:
Let's push on, merging this and tagging a new release 👍
